release notes/v0.40.0.md

```diff
@@ -0,0 +1,285 @@
+
+Additionally, as previously mentioned, there were a bunch of changes to goja, including adding class support which also needed to be integrated.
+
+A future breaking change is that using CommonJS style exports along import/export syntax in the *same* file will no longer be possible.
+
+```javascript
+import http from "k6/http"
+
+exports.default = function() {} // this will start to error out
+```
+
+It will still be possible to import a CommonJS module and `require` an ES module or use `require` in an ES module. Having a dependency cycle mixing and matching CommonJS and ESM is also unlikely to work properly, but *might* do so in particular cases.
+
+This is really expected to have never been done by anyone as there isn't really much of a reason for this, but it is currently supported due to Babel transpiling everything to CommonJS behind the scenes.
+
+### Refactoring metrics
+
+The refactoring of metrics is underway, with a [PR](https://github.com/grafana/k6/pull/2594) for more performant internal representation. Unfortunately, this took longer to get to a stable state, and given the size and the type of change, the k6 team decided to *hold* merging it until very near the end of the cycle. It also doesn't have any noticeable change for most users. Instead, it will be merged early in the v0.41.0 release cycle, and then more changes to use it will be made through the release cycle.
+
+Some of those changes include supporting non-indexable tags, which also have a read [PR](https://github.com/grafana/k6/pull/2654). This change is also essential for historical reasons connected with how the `name`  tag works. As such, it also needed to be merged to release the internal metric refactor.
+
+Future breaking change: as part of the many changes and experiments, we found out that we can keep `url` as an indexable tag. Previously the plan was that it *along* `vu` and `iter` will become non-indexable. However, the `url` tag is *heavily* used and enabled by default. Because of that (and other internal reasons), the new change will be that `url` will stay indexable, but if used with `name` will be overwritten to have the same value as `name`. Arguably this is what most users would want in the case when they are using `name`. We plan to add a *non* indexable `raw_url` tag for those that do not. As such, we no longer will be [printing a warning when `url` is used](https://github.com/grafana/k6/pull/2655) in thresholds.
+
+Even so, we did make a bunch of changes to the internals of k6 that will pave the way forward ([#2629](https://github.com/grafana/k6/pull/2629), [#2631](https://github.com/grafana/k6/pull/2631)).
+
+[Prometheus remote-write output extension](https://github.com/grafana/xk6-output-prometheus-remote/) was on hold as it heavily depends on metrics' refactoring.
```

Comment:
I am for a shortened version of @codebien's suggestion. We don't need to give that many details in the release notes' roadmap. Something like this maybe?
```suggestion
We are also still working on incorporating the newly developed time series data model for the [Prometheus remote-write output extension](https://github.com/grafana/xk6-output-prometheus-remote). We are fixing bugs and improving the extension with the goal of eventually integrating it as a core built-in k6 output module in a future k6 release.
```

Reply:
:man_shrugging: Not really certain how much information we should be providing - I would definitely agree with you on it being shorter, if there was `xk6-output-prometheus-remote` release notes that we can link to :shrug:

Reply:
Maybe the updates on different extensions that won't be coming in the core or the core k6 doesn't do something specific for should just be in a blog post? This will also include xk6-browser. We do already have a bunch of blog posts, emails and community forum threads that come out after the release.

Having a "this month in k6" kind of newsletter is likely going to be beneficial with us now having a bunch of different projects with differetent release cycles (at least for now). 

And this will mean that we won't have to keep expanding the k6 release notes with release notes for other extensions. 

But also users won't need to look at 10 different release notes for 10 different things.

Reply:
I'd join @na-- in his shorter version since it matches the "Roadmap and future plans" then just `Prometheus remote-write output extension was on hold as it heavily depends on metrics' refactoring.` 

I mean the current version is not plan it's a fact :sweat_smile:

Reply:
We included it in the previous release notes so I think it's good to provide an update on the current status. I'm fine with having a very short version.

>we did some specific work on this one extension that we know isn't going to stick this way and likely would have changed by the time you look at it".

Just to mention, this was true before, but now flushing by time series for the end-user when we will change for the new time series model it wouldn't be impacted.